tasks/kmt.py

```diff
@@ -695,6 +692,30 @@ def prepare(
             f"Architecture {arch} (inferred {arch_obj}) is not supported. Supported architectures are amd64 and arm64"
         )
 
+    domains = get_target_domains(ctx, stack, ssh_key, arch_obj, vms, alien_vms)
+    used_archs = get_archs_in_domains(domains)
+
+    if alien_vms is not None:
+        err_msg = f"no alient VMs discovered from provided profile {alien_vms}."
+    else:
+        err_msg = f"no vms found from list {vms}. Run `inv -e kmt.status` to see all VMs in current stack"
+
+    assert len(domains) > 0, f"no vms found from list {vms}. Run `inv -e kmt.status` to see all VMs in current stack"
+
+    _prepare(ctx, component, domains, stack, arch, packages, verbose, ci, compile_only)
```

Comment:
fix: the arguments for this call are switched

```suggestion
    _prepare(ctx, stack, component, domains, arch, packages, verbose, ci, compile_only)
```